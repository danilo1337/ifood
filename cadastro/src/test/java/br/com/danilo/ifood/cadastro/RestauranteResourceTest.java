package br.com.danilo.ifood.cadastro;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestauranteResourceTest {
	
	@Test
	@DataSet("restaurantes-cenario1.yml")
	public void testBuscarRestaurantes() {
		String resultado = RestAssured.given()
				.when().get("/restaurantes")
				.then()
				.statusCode(200)
				.extract().asString();
		Approvals.verifyJson(resultado);
	}
		
}
