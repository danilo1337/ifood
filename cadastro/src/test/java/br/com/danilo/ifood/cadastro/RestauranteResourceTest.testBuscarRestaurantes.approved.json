[
  {
    "id": 123,
    "nome": "Novo Restaurante",
    "propietario": "ID do keyclock"
  },
  {
    "id": 234,
    "nome": "Novo Restaurante 2",
    "propietario": "ID do keyclock 2"
  }
]